pkg/auth/oauth/handlers/default_auth_handler.go

```diff
@@ -14,25 +14,28 @@ import (
 
 // unionAuthenticationHandler is an oauth.AuthenticationHandler that muxes multiple challenge handlers and redirect handlers
 type unionAuthenticationHandler struct {
-	challengers      map[string]AuthenticationChallenger
-	redirectors      map[string]AuthenticationRedirector
-	errorHandler     AuthenticationErrorHandler
-	selectionHandler AuthenticationSelectionHandler
+	challengers        map[string]AuthenticationChallenger
+	redirectors        map[string]AuthenticationRedirector
```

Comment:
instead of parallel structures, I'd rather see a type that handles the ordering and lookup internally... something like
```
type AuthenticationRedirectors interface {
  Add(name string, redirector AuthenticationRedirector)
  Get(name string) (redirector AuthenticationRedirector, exists bool)
  Names() []string
}
```
impl could just be a map+list internally